internal/cloud/backend.go

```diff
@@ -178,18 +183,26 @@ func (b *Cloud) PrepareConfig(obj cty.Value) (cty.Value, tfdiags.Diagnostics) {
 	}
 
 	WorkspaceMapping := WorkspaceMapping{}
+
+	// Initially set the project via env var
+	WorkspaceMapping.Project = os.Getenv("TF_CLOUD_PROJECT")
+
+	// Initially set the workspace name via env var
+	WorkspaceMapping.Name = os.Getenv("TF_WORKSPACE")
+
 	if workspaces := obj.GetAttr("workspaces"); !workspaces.IsNull() {
 		if val := workspaces.GetAttr("name"); !val.IsNull() {
 			WorkspaceMapping.Name = val.AsString()
 		}
+		if val := workspaces.GetAttr("project"); !val.IsNull() {
+			WorkspaceMapping.Project = val.AsString()
+		}
 		if val := workspaces.GetAttr("tags"); !val.IsNull() {
 			err := gocty.FromCtyValue(val, &WorkspaceMapping.Tags)
 			if err != nil {
 				log.Panicf("An unxpected error occurred: %s", err)
```

Comment:
Oops, I see a typo! I know it's not your commit, but could you change that "unxpected" for "unexpected"? 🙏